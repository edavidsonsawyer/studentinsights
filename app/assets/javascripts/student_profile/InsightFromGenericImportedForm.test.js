import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {withDefaultNowContext} from '../testing/NowContainer';
import InsightFromGenericImportedForm from './InsightFromGenericImportedForm';


function testProps(props = {}) {
  return {
    student: {
      id: 9,
      first_name: 'Mari',
      last_name: 'Kenobi'
    },
    insightPayload: {
      prompt_text: "At the end of the quarter 3, what would make you most proud of your accomplishments in your course?",
      response_text: "Keeping grades high in all classes since I'm worried about college",
      flattened_form_json: {
        form_title: 'Q2 Self-reflection',
        form_timestamp: '2018-03-06T11:03:00.000Z',
        text: '<the whole survey text>'
      }
    },
    ...props
  };
}

function testEl(props = {}) {
  return withDefaultNowContext(<InsightFromGenericImportedForm {...props} />);
}


// since element sizing methods aren't present in the Jest JS environment
jest.mock('../components/FitText', () => 'mocked-fit-text');

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(testEl(props), el);
});

it('snapshots view', () => {
  const props = testProps();
  const tree = renderer
    .create(testEl(props))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
