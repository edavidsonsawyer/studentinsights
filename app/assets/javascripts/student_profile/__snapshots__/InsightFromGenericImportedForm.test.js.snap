// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div
  className="InsightQuote InsightFromStudent "
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "margin": 20,
        "marginBottom": 0,
        "marginTop": 15,
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <div
          style={
            Object {
              "fontSize": 12,
            }
          }
        >
          At the end of the quarter 3, what would make you most proud of your accomplishments in your course?
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "marginTop": 5,
            }
          }
        >
          <mocked-fit-text
            fontSizeStep={4}
            maxFontSize={48}
            minFontSize={12}
            text="“Keeping grades high in all classes since I'm worried about college”"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 12,
        "justifyContent": "space-between",
        "margin": 20,
        "marginBottom": 15,
        "marginTop": 10,
      }
    }
  >
    <div>
      <div>
        <div>
          from 
          <a
            href="/students/9"
            rel="noopener noreferrer"
            style={
              Object {
                "fontSize": 12,
              }
            }
            target="_blank"
          >
            Mari
             
            Kenobi
          </a>
        </div>
        <div>
          <div>
            <span>
              in 
            </span>
            <div
              style={
                Object {
                  "display": "inline",
                  "margin": 0,
                  "marginLeft": 10,
                }
              }
            >
              <a
                href="#"
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": 12,
                    "outline": "none",
                  }
                }
              >
                Q2 Self-reflection
              </a>
            </div>
            <span>
               on 
              3/6/18
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
